### PR TITLE
Fixed #31044 -- Errored nicely when using Prefetch with a raw() queryset.

### PR DESCRIPTION
**Pull request by hramezani**
[Ticket 31044](https://code.djangoproject.com/ticket/31044)

### CODE REVIEW

File: django/db/models/query.py [hunk 1]
```diff
@@ -1534,8 +1534,11 @@ def __init__(self, lookup, queryset=None, to_attr=None):
         self.prefetch_through = lookup
         # `prefetch_to` is the path to the attribute that stores the result.
         self.prefetch_to = lookup
-        if queryset is not None and not issubclass(queryset._iterable_class, ModelIterable):
-            raise ValueError('Prefetch querysets cannot use values().')
+        if queryset is not None:
+            if not hasattr(queryset, '_iterable_class'):
+                raise AttributeError("Prefetch queryset should has an '_iterable_class' attribute.")
```
Comment (felixxm) [inline django/db/models/query.py]: We shouldn't raise errors about private APIs, we can error nicely when using `Prefetch` with `raw()`, e.g.
```python
if queryset is not None:
    if isinstance(queryset, RawQuerySet) or (
        hasattr(queryset, '_iterable_class') and not issubclass(queryset._iterable_class, ModelIterable)
    ):
        raise AttributeError('Prefetch querysets cannot use raw() and values().')
```

Comment (hramezani) [reply django/db/models/query.py]: replaced you suggested patch and changed `AttributeError` with `ValueError`.

Comment (felixxm) [reply django/db/models/query.py]: Yes, `ValueError` sorry.

**Comment by felixxm** [comment]
@hramezani Thanks :+1: I pushed minor edits and fixed an error message for `Prefetch()` with a `values_list()` queryset.

**Pull request merged**